lib/grandpa/message_handler.go

```diff
@@ -284,27 +284,32 @@ func (h *MessageHandler) verifyCatchUpResponseCompletability(prevote, precommit
 	return nil
 }
 
-func getEquivocatoryVoters(votes []AuthData) map[ed25519.PublicKeyBytes]struct{} {
+func getEquivocatoryVoters(votes []AuthData) (map[ed25519.PublicKeyBytes]struct{}, error) {
 	eqvVoters := make(map[ed25519.PublicKeyBytes]struct{})
 	voters := make(map[ed25519.PublicKeyBytes]int, len(votes))
 
 	for _, v := range votes {
 		voters[v.AuthorityID]++
+		if voters[v.AuthorityID] > 2 {
+			return nil, errInvalidMultiplicity
```

Comment:
nit perhaps wrap some context information?

```suggestion
			return nil, fmt.Errorf("%w: authority id %x has %d votes", 
				errInvalidMultiplicity, v.AuthorityID, voters[v.AuthorityID])
```